src/renderer/features/fellowship-evidence-salary/model/evidenceVoting.ts

```diff
@@ -0,0 +1,194 @@
+import { type ApiPromise } from '@polkadot/api';
+import { combine, createEffect, createEvent, createStore, restore, sample } from 'effector';
+import { createGate } from 'effector-react';
+import { reshape } from 'patronum';
+
+import { nonNullable, nonNullableMap, nullable } from '@/shared/lib/utils';
+import { type ReferendumId, referendaPallet } from '@/shared/pallet/referenda';
+import { createTxStore } from '@/shared/transactions';
+import { type Evidence, votingService } from '@/domains/collectives';
+import { type SigningPayload, signModel } from '@/features/operations/OperationSign';
+import { submitModel } from '@/features/operations/OperationSubmit';
+
+import { fellowshipSalaryFeature } from './feature';
+import { fellowship } from './fellowship';
+
+const flow = createGate<{ evidence: Evidence | null; aye: boolean }>({ defaultState: { evidence: null, aye: false } });
+
+const { $api, $chain, $wallet, $wallets, $votingAccount, $votingMember } = reshape({
+  source: fellowshipSalaryFeature.input,
+  shape: {
+    $api: x => x?.api ?? null,
+    $wallets: x => x?.wallets ?? [],
+    $wallet: x => x?.wallet ?? null,
+    $chain: x => x?.chain ?? null,
+    $votingAccount: x => x?.account ?? null,
+    $votingMember: x => x?.member ?? null,
+  },
+});
+
+const $members = fellowship.$store.map(s => s?.members ?? []);
+const $tracks = fellowship.$store.map(s => s?.tracks ?? []);
+const $maxRank = fellowship.$store.map(s => s?.maxRank ?? 0);
+
+const $evidence = flow.state.map(s => s.evidence);
+const $aye = flow.state.map(s => s.aye);
+
+const $member = combine($members, $evidence, (members, evidence) => {
+  return members.find(m => m.accountId === evidence?.accountId) ?? null;
+});
+
+const $currentTrack = combine($member, $tracks, (member, tracks) => {
+  if (nullable(member)) return null;
+  return tracks.find(t => t.id === member.rank) ?? null;
+});
+
+const $nextTrack = combine($member, $tracks, (member, tracks) => {
+  if (nullable(member)) return null;
+  return tracks.find(t => t.id === member.rank + 1) ?? null;
+});
+
+const $nextReferendum = createStore<ReferendumId | null>(null);
+
+const requestNextReferendumFx = createEffect(({ api }: { api: ApiPromise }) => {
+  return referendaPallet.storage
+    .referendumCount('fellowship', api)
+    .then(c => c + 1)
```

Comment:
I assume this should work fine
```suggestion
    .then(c => referendaPallet.helpers.toReferendumId(c + 1))
```